lib/Ed25519VerificationKey2020.js

```diff
@@ -31,10 +39,24 @@ export class Ed25519VerificationKey2020 extends LDKeyPair {
     super(options);
     this.type = SUITE_ID;
     this.publicKeyMultibase = options.publicKeyMultibase;
+    this.privateKeyMultibase = options.privateKeyMultibase;
+
     if(!this.publicKeyMultibase) {
       throw new TypeError('The "publicKeyMultibase" property is required.');
     }
-    this.privateKeyMultibase = options.privateKeyMultibase;
+
+    if(!this.publicKeyMultibase || !_isValidKeyHeader(
+      this.publicKeyMultibase, MULTICODEC_ED25519_PUB_HEADER)) {
+      throw new Error('`publicKeyMultibase` has invalid byte header: ' +
+      `"${this.publicKeyMultibase}".`);
+    }
+
+    if(this.privateKeyMultibase && !_isValidKeyHeader(
+      this.privateKeyMultibase, MULTICODEC_ED25519_PRIV_HEADER)) {
+      throw new Error('`privateKeyMultibase` has invalid byte header: ' +
+       `"${this.privateKeyMultibase}".`);
+    }
```

Comment:
I find it really odd to set things on `this` and *then* validate the inputs. It's also not in line with our standard practices. Let's validate `options.*` and then set them on `this` if everything checks out, i.e. do validation of the options at the very top of the function.

Reply:
Fixed in 2db6cc7ae7988dde2660d438b538777442a1c67a.